pkg/ebpf/signature_engine.go

```diff
@@ -19,6 +20,9 @@ func (t *Tracee) engineEvents(ctx context.Context, in <-chan *trace.Event) (<-ch
 	engineOutput := make(chan detect.Finding, 100)
 	engineInput := make(chan protocol.Event)
 	source := engine.EventSources{Tracee: engineInput}
+
+	t.config.EngineConfig.DataSources = append(t.config.EngineConfig.DataSources, containers.NewDataSource(t.containers))
```

Comment:
Why initialize it here? I feel like this should be done at the creation of the engine.Config, and not hidden inside the pipeline.

Reply:
I'll see if I can make this move work.

Reply:
I came here for the same reason =D.

Reply:
So the data source implementation uses tracee's internal container tracker `(*tracee).containers`.
We create the EngineConfig in `GetTraceeRunner`, where the `tracee` object isn't created yet. As such I can't create them at the same time.
Not sure if a refactor to make this possible is worth it, at least right now.

Reply:
Fair enough.